Authentication events policy docs for user flow invocation (#10224)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/authenticationaction.md

---
title: "authenticationAction resource type"
description: "Defines the action to take during an authentication event."
author: "jkdouglas"
localization_priority: Normal
ms.prod: "microsoft-identity-platform"
doc_type: resourcePageType
---

# authenticationAction resource type

Namespace: microsoft.graph

Defines the action to take during an authentication event. An authenticationAction is abstract and is the base class of other actions you can take during an authentication event. You can invoke a user flow in the onSignUpStart authentication event contained in an [authenticationEventsPolicy](../resources/authenticationeventspolicy.md). To invoke a user flow, you use the [invokeUserFlowAction](../resources/invokeuserflowaction.md) resource.

## Methods

|Method|Return type|Description|
|:---|:---|:---|

> |[List authenticationActions](../api/authenticationaction-list.md)|[authenticationAction](../resources/authenticationaction.md) collection|Get the authenticationAction objects for a specific authentication event.|

Reviewer: Lines 20-21:
Please update the Method column topic title, link destination, and the Description column description, per my comments in what you currently call authenticationaction-list.md, and authentication-create.md. (Note I have commented there to rename those 2 filenames accordingly.)

|[Create authenticationAction](../api/authenticationeventspolicy-post-onsignupstart.md)|[authenticationAction](../resources/authenticationaction.md)|Create a new authenticationAction object for the onSignupStart event.|

> |[Update authenticationAction](../api/authenticationaction-update.md)|[authenticationAction](../resources/authenticationaction.md)|Update the properties of an authenticationAction object.|

Reviewer: Line 22:
Per my comment in authentication-action.md, can we update the description to give some more context, for example:
"Update the specified action defined for an event in the authentication pipeline."

> |[Get authenticationAction](../api/authenticationaction-get.md)|[authenticationAction](../resources/authenticationaction.md)|Read the properties and relationships of an authenticationAction object.|

Reviewer: Line 23:
Per my comment in authenticationaction-get.md, can we update the description to give some more context, for example:
"Get the specified action defined for an event in the authentication pipeline."

> |[Remove authenticationAction](../api/authenticationaction-remove.md)|None|Remove an [authenticationAction](../resources/authenticationaction.md) object.|

Reviewer: Line 24:
The Method column should say "Delete authenticationAction".

Reviewer: Line 24:
The Description column should also be consistent and say "Delete an ..."

> |[Remove authenticationAction](../api/authenticationaction-remove.md)|None|Remove an [authenticationAction](../resources/authenticationaction.md) object.|

Reviewer: Line 23:
Per my comment in the authentication-get.md topic, please update the description to be the same:
"Get the specified action defined for an event in the authentication pipeline."

> |[Delete authenticationAction](../api/authenticationaction-delete.md)|None|Delete an authenticationAction from an event supported by an authenticationEventsPolicy.|

Reviewer: Line 24:
Please change the description to:
"Delete the specified action defined for the onSignupStart event in the authentication pipeline."

---
title: "authenticationAction resource type"
description: "Defines the action to take during an authentication event."
author: "jkdouglas"
localization_priority: Normal
ms.prod: "microsoft-identity-platform"
doc_type: resourcePageType
---

# authenticationAction resource type

Namespace: microsoft.graph

Defines the action to take during an authentication event. An authenticationAction is abstract and is the base class of other actions you can take during an authentication event. You can invoke a user flow in the onSignUpStart authentication event contained in an [authenticationEventsPolicy](../resources/authenticationeventspolicy.md). To invoke a user flow, you use the [invokeUserFlowAction](../resources/invokeuserflowaction.md) resource.

## Methods

|Method|Return type|Description|
|:---|:---|:---|

> |[List onSignupStart](../api/authenticationeventspolicy-list-onsignupstart.md)|[authenticationAction](../resources/authenticationaction.md) collection|Get the authenticationAction resources from the onSignupStart navigation property.|

Reviewer 2: ```suggestion
|[List onSignupStart](../api/authenticationeventspolicy-list-onsignupstart.md)|[authenticationAction](../resources/authenticationaction.md) collection|Get the authenticationActions from the onSignupStart object.|
```

|[Create onSignupStart](../api/authenticationeventspolicy-post-onsignupstart.md)|[authenticationAction](../resources/authenticationaction.md)|Create a new authenticationAction object.|
|[Update onSignupStart](../api/authenticationeventspolicy-update-onsignupstart.md)|[authenticationAction](../resources/authenticationaction.md)|Update the properties of an onSignupStart object.|
|[Get onSignupStart](../api/authenticationeventspolicy-get-authenticationaction.md)|[authenticationAction](../resources/authenticationaction.md)|Read the properties and relationships of an [authenticationAction](../resources/authenticationaction.md) object.|
|[Delete onSignupStart](../api/authenticationeventspolicy-delete-onsignupstart.md)|None|Delete an [authenticationAction](../resources/authenticationaction.md) object.|

## Properties

|Property|Type|Description|
|:---|:---|:---|
|id|String|The identifier of the action.|

> |priority|Int32|The priority of the action which is used for determining one out of multiple applicable actions.|

Reviewer 2: ```suggestion
|priority|Int32|The priority of the action that is used to determine one out of multiple applicable actions.|
```

> |sourceFilter|[authenticationSourceFilter](../resources/authenticationsourcefilter.md)|Filter based on the source of the authentication which is used to determine whether the action is executed or not.|

Reviewer 2: ```suggestion
|sourceFilter|[authenticationSourceFilter](../resources/authenticationsourcefilter.md)|Filter based on the source of the authentication that is used to determine whether the action is executed.|
```
